src/framework/components/element/text-element.js

```diff
@@ -482,37 +641,25 @@ class TextElement {
                 }
 
                 this._setTextureParams(mi, this._font.textures[i]);
-                if (this._symbolColors) {
-                    // when per-vertex coloring is present, disable material emissive color
-                    this._colorUniform[0] = 1;
-                    this._colorUniform[1] = 1;
-                    this._colorUniform[2] = 1;
-                } else {
-                    this._colorUniform[0] = this._color.r;
-                    this._colorUniform[1] = this._color.g;
-                    this._colorUniform[2] = this._color.b;
-                }
+
                 mi.setParameter('material_emissive', this._colorUniform);
                 mi.setParameter('material_opacity', this._color.a);
                 mi.setParameter('font_sdfIntensity', this._font.intensity);
                 mi.setParameter('font_pxrange', this._getPxRange(this._font));
                 mi.setParameter('font_textureWidth', this._font.data.info.maps[i].width);
 
-                this._outlineColorUniform[0] = this._outlineColor.r;
-                this._outlineColorUniform[1] = this._outlineColor.g;
-                this._outlineColorUniform[2] = this._outlineColor.b;
-                this._outlineColorUniform[3] = this._outlineColor.a;
                 mi.setParameter('outline_color', this._outlineColorUniform);
                 mi.setParameter('outline_thickness', this._outlineThicknessScale * this._outlineThickness);
 
-                this._shadowColorUniform[0] = this._shadowColor.r;
-                this._shadowColorUniform[1] = this._shadowColor.g;
-                this._shadowColorUniform[2] = this._shadowColor.b;
-                this._shadowColorUniform[3] = this._shadowColor.a;
                 mi.setParameter('shadow_color', this._shadowColorUniform);
-                const ratio = -this._font.data.info.maps[i].width / this._font.data.info.maps[i].height;
-                this._shadowOffsetUniform[0] = this._shadowOffsetScale * this._shadowOffset.x;
-                this._shadowOffsetUniform[1] = ratio * this._shadowOffsetScale * this._shadowOffset.y;
+                if (this._symbolShadowParams) {
+                    this._shadowOffsetUniform[0] = 0;
+                    this._shadowOffsetUniform[0] = 0;
```

Comment:
Should it be `_shadowOffsetUniform[1]`?

Reply:
You are right. It can break batching. Thanks!
Fixed